File: frontend/src/types/media.ts
export interface SubtitleTrack {
  id: number;
  language: string;
  is_source: boolean;
  status: string;
  error_message: string | null;
}

export interface MediaFile {
  id: number;
  filename: string;
  media_type: string;
  file_size: number;
  duration: number;
  format: string;
  source_language: string | null;
  transcription_status: string;
  error_message: string | null;
  upload_time: string;
  subtitle_tracks: SubtitleTrack[];
}

export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface CreateMediaResponse {
  media_id: number;
  upload_url: string;
  s3_key: string;
}

// Common target languages offered in the multi-select. value = code sent to the
// backend (used both as the SubtitleTrack.language and the translation target).
export const LANGUAGE_OPTIONS: { code: string; label: string }[] = [
  { code: "en", label: "English" },
  { code: "zh", label: "中文" },
  { code: "ja", label: "日本語" },
  { code: "ko", label: "한국어" },
  { code: "es", label: "Español" },
  { code: "fr", label: "Français" },
  { code: "de", label: "Deutsch" },
  { code: "pt", label: "Português" },
  { code: "ru", label: "Русский" },
  { code: "ar", label: "العربية" },
];

File: backend/app/schemas/media.py
from __future__ import annotations
from datetime import datetime

from pydantic import BaseModel


class SubtitleTrackResponse(BaseModel):
    id: int
    language: str
    is_source: bool
    status: str
    error_message: str | None = None

    model_config = {"from_attributes": True}


class MediaFileResponse(BaseModel):
    id: int
    filename: str
    media_type: str
    file_size: int
    duration: float
    format: str
    source_language: str | None = None
    transcription_status: str
    error_message: str | None = None
    upload_time: datetime
    subtitle_tracks: list[SubtitleTrackResponse] = []

    model_config = {"from_attributes": True}


class CreateMediaRequest(BaseModel):
    filename: str
    content_type: str | None = None


class CreateMediaResponse(BaseModel):
    media_id: int
    upload_url: str
    s3_key: str


class GenerateSubtitlesRequest(BaseModel):
    target_languages: list[str]


class BatchGenerateRequest(BaseModel):
    media_ids: list[int]
    target_languages: list[str]
